design/m1_push_to_file_design.md

```diff
@@ -0,0 +1,1163 @@
+
+##### Supported Annotations
+
+| Annotation                                     | Description |
+|------------------------------------------------|-------------|
+| conjur.org/conjur-secrets.{secret-group}       | List of secrets to be retrieved from Conjur. Each entry can be either:<ul><li>A Conjur variable path</li><li> A key/value pairs of the form `<alias>:<Conjur variable path>` where the `alias` represents the name of the secret to be written to the secrets file |
+| conjur.org/conjur-secrets-path.{secret-group}  | Prefix to use for all Conjur variable paths for a secrets group. Defaults to root policy path. |
+| conjur.org/secret-file-path.{secret-group}     | File path (including file name) for secrets file to be written. Default is `{secret-group}.{secret-file-type}`, e.g. `database.yaml`. |
+| conjur.org/secret-file-type.{secret-group}     | Allowed values:<ul><li>yaml (default)</li><li>json</li><li>dotenv</li><li>bash</li></ul>This setting is ignored when `conjur.org/secret-file-template.{secret-group}` is configured. |
+| conjur.org/secret-file-perms.{secret-group}    | File permissions for secrets file to be written, in octal format. Defaults to `660` |
+| conjur.org/secret-file-template.{secret-group} | Customized secret file template using the Golang templating language. |
+
+###### Example `conjur-secrets` List Without Aliases
+
+The following is an example of a `conjur.org/conjur-secrets.{secret-group}`
+Annotation that defines a flat (i.e. no aliases) list of Conjur variable paths
+for a hypothetical secrets group named `database`:
+
+```
+    conjur.org/conjur-secrets.database: |
+      - prod/backend/url
+      - prod/backend/port 
+      - prod/backend/password
+      - prod/backend/username
+```
+
+For `conjur-secrets` entries without aliases, the last word in the path is
+used as the application secret name. So for the example above, if
+`conjur.org/secret-file-type.database` is set to `yaml`, then the resulting
+secrets file that is written to the shared secrets volume might look similar
+to the following:
+
+```
+url: https://database.example.com
+port: 5
+password: my-secret-p@$$w0rd
+username: postgres
+```
+
+###### Example `conjur-secrets` List With Aliases and `conjur-secrets-path` Prefix
+
+The following is an example of a `conjur.org/conjur-secrets.{secret-group}`
+Annotation that defines a list of Conjur variable paths, including some that
+use an alias, for a hypothetical secrets group named `cache`:
+
+```
+    conjur.org/conjur-secrets.cache: |
+      - url
+      - admin-password: password
+      - admin-username: username
+
+    conjur.org/conjur-secrets-path.cache: dev/memcached/
+
+```
+
+In this example, the paths for the secrets that are retrieved from Conjur are:
+
+- dev/memcached/url
+- dev/memcached/password
+- dev/memcached/username
+
+and the names of the secrets that are written to the secrets file are:
+
+- url
+- admin-password
+- admin-username
+
```

Comment:
we could add here:

---

so that the `cache.yaml` file that Secrets Provider writes includes:
```
url: https://database.example.com
admin-password: my-secret-p@$$w0rd
admin-username: admin
```